engine/classes/Elgg/Views/ViewtypeRegistry.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Elgg\Views;
+
+use Elgg\Structs\ArrayCollection;
+use Elgg\Structs\Map;
+use Elgg\Http\Input;
+
+
+/**
+ * Manages the list of registered views as well as the current system viewtype.
+ * 
+ * @since 2.0.0
+ * @access private
+ */
+class ViewtypeRegistry implements Map/*<string,Viewtype>*/ {
+	/** @var \stdClass */
+	private $config;
+	
+	/** @var Viewtype[] */
+	private $items;
+	
+	/** @var Input */
+	private $input;
+	
+	/** @var ?Viewtype */
+	private $current;
+	
+	/**
+	 * Constructor
+	 * 
+	 * @param \stdClass $config The Elgg config object
+	 * @param Input     $input  Request inputs
+	 */
+	public function __construct(\stdClass $config, Input $input) {
+		$this->config = $config;
+		$this->input = $input;
```

Comment:
#8438 would eliminate the need to have this depend on Input. Just pass the value in.